src/app/devices/deviceContent/components/cloudToDeviceMessage/cloudToDeviceMessage.tsx

```diff
@@ -64,387 +64,368 @@ export interface CloudToDeviceMessageProps {
-                        checkButtonAriaLabel={context.t(ResourceKeys.cloudToDeviceMessage.properties.rowCheckBoxAriaLabel)}
-                        selection={this.state.selection}
-                    />
-                </MarqueeSelection>
-            </>
-        );
-    }
-
-    private readonly onDropdownSelectedKeyChanged = (property: PropertyItem) => (event: React.FormEvent<HTMLDivElement>, option?: IDropdownOption): void => {
-        const items = this.state.properties;
-        const index = this.findMatchingItemIndex(property);
+    const onDropdownSelectedKeyChanged = (property: PropertyItem) => (event: React.FormEvent<HTMLDivElement>, option?: IDropdownOption): void => {
+        const items = properties;
+        const index = findMatchingItemIndex(property);
         items[index] = {...items[index], value: option.key.toString()};
-        this.setState({properties: items});
-    }
-
-    private readonly onSelectionChanged = () => {
-        this.setState({
-            selectedIndices: new Set(this.state.selection.getSelectedIndices())
-        });
-    }
-
-    private readonly handleAddCustomProperty = () => {
-        const newIndex = this.state.propertyIndex + 1;
-        this.setState(prevState => ({
-            properties: [...prevState.properties, {isSystemProperty: false, index: newIndex, keyName: '', value: ''}],
-            propertyIndex: newIndex
-        }));
-    }
-
-    private readonly handleAddSystemProperty = (keyName: string) => () => {
-        const newIndex = this.state.propertyIndex + 1;
-        this.setState(prevState => ({
-            properties: [...prevState.properties, {isSystemProperty: true,  index: newIndex, keyName, value: ''}],
-            propertyIndex: newIndex
-        }));
-    }
-
-    private readonly handleEditCustomPropertyKey = (property: PropertyItem) => ((event: React.FormEvent<HTMLInputElement | HTMLTextAreaElement>, newValue?: string) => {
-        const items = this.state.properties;
-        const index = this.findMatchingItemIndex(property);
+        setProperties(items);
+    };
+
+    const onSelectionChanged = () => {
+        setSelectedIndices(new Set(selection.getSelectedIndices()));
+    };
+
+    const handleAddCustomProperty = () => {
+        const newIndex = propertyIndex + 1;
+        const newProperties = [...properties, {isSystemProperty: false, index: newIndex, keyName: '', value: ''}];
+        setProperties(newProperties);
+        setPropertyIndex(newIndex);
+    };
+
+    const handleAddSystemProperty = (keyName: string) => () => {
+        const newIndex = propertyIndex + 1;
+        const newProperties = [...properties, {isSystemProperty: true,  index: newIndex, keyName, value: ''}];
+        setProperties(newProperties);
+        setPropertyIndex(newIndex);
+    };
+
+    const handleEditCustomPropertyKey = (property: PropertyItem) => ((event: React.FormEvent<HTMLInputElement | HTMLTextAreaElement>, newValue?: string) => {
```

Comment:
this function along with handleEditPropertyValue, doesn't get triggered while user is typing in the input, but only after user click away somehow. We need to take a closer look here